blockbase/sass/base/_alignment.scss

```diff
@@ -1,17 +1,36 @@
-.wp-block-group.alignfull,
-*[class*="wp-container-"] //Anything that inherits layout (container)
-{
-	//give it some padding
-	padding-left: var(--wp--custom--post-content--padding--left);
-	padding-right: var(--wp--custom--post-content--padding--right);
+//FRONTEND
+.blockbase-alignment-rules, //for argeted elements
+.wp-block-post>.wp-block-group, //for group blocks in the query loop
+.wp-block-post-content { //for the post content (NOTE: margins for this block cannot be configured via theme.json at this time.)
+	//In this situation we want to introduce a standardized gap between content and the edge of the screen.
+	margin-left: var(--wp--custom--post-content--padding--left);
+	margin-right: var(--wp--custom--post-content--padding--right);
+	.alignfull {
+		// these elements we want to "bust out" of the gap created above 
+		margin-left: calc(-1 * var(--wp--custom--post-content--padding--left)) !important;
+		margin-right: calc(-1 * var(--wp--custom--post-content--padding--right)) !important;
+		width: unset;
+		// however any groups that "bust out" should re-apply that gap but this time using padding instead of margins.
+		&.wp-block-group {
+	 		padding-left: var(--wp--custom--post-content--padding--left);
+	 		padding-right: var(--wp--custom--post-content--padding--right);
+		}
+	}
+}
 
-	//Any nested containers, and anything that is alignfull
-	*[class*="wp-container-"], // Any nested containers
-	> .alignfull { // Any direct descendant that is alignfull
-		// bust out of the container's padding
+// EDITOR (NOTE: It PROBABLY would be OK to bring these together to "simplify" the stylesheet.  However the selectors are quite different
+// and it's a lot easier to understand and ensure intent separated in this way.
+.is-root-container { //top level of the editor
```

Comment:
does this also work in the site editor?

Reply:
Yes.  We looked carefully at the editor.  It seems to cover the same use-cases as the view.